vignettes/adcoeq.Rmd

```diff
@@ -0,0 +1,261 @@
+
+**Note**: *All examples assume CDISC SDTM and/or ADaM format as input unless otherwise specified.*
+
+## Required Packages
+
+The examples of this vignette require the following packages.
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+#library(admiralmetabolic)
+devtools::load_all()
+library(pharmaversesdtm)
+library(dplyr)
+```
+
+# Programming Workflow
+
+-   [Read in Data](#readdata)
+-   [Original items](#original_items)
+-   [Derive the four Subscales](#subscales)
+-   [Derive baseline and change from baseline](#chg)
+
+
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of the ADaM dataset should be loaded into the global environment. Reading data will usually be a company specific process, however, for the purpose of this vignette, we will use example data from `{pharmaversesdtm}` and `{admiral}`. We will utilize `DM`, `QS` and `ADSL`.
+
+
+In this vignette we use example data for the CoEQ. 
+The example `QS` data (`qs_metabolic`) is included in the `{admiralmetabolic}` package.
+
+```{r, message=FALSE, warning=FALSE}
+dm_metabolic <- admiralmetabolic::dm_metabolic
+qs_metabolic <- admiralmetabolic::qs_metabolic
+admiral_adsl <- admiral::admiral_adsl
+
+dm <- convert_blanks_to_na(dm_metabolic)
+qs <- convert_blanks_to_na(qs_metabolic)
+admiral_adsl <- convert_blanks_to_na(admiral_adsl)
+```
+
+Within this vignette, `DM` is used as the basis for `ADSL`:
+
+```{r echo=FALSE}
+dm <- admiralmetabolic::dm_metabolic
+
+adsl <- dm %>%
+  select(-DOMAIN) %>%
+  mutate(TRT01P = ARM, TRT01A = ACTARM) %>%
+  left_join(admiral_adsl %>% select(USUBJID, TRTSDT, TRTEDT), by = "USUBJID")
+```
+
+## Original Items {#original_items}
+
+The original items, i.e. the answers to the questionnaire questions, can be handled in the same way as in a [{admiral} BDS finding ADaM](https://pharmaverse.github.io/admiral/articles/bds_finding.html). 
+
+```{r eval=TRUE}
+adcoeq1 <- qs %>%
+  # Add ADSL variables
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = exprs(TRTSDT, TRTEDT, TRT01P, TRT01A),
+    by_vars = exprs(STUDYID, USUBJID)
+  ) %>%
+  # Add analysis parameter variables
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = QSTEST,
+    PARCAT1 = QSCAT
+  ) %>%
+  # Add timing variables
+  derive_vars_dt(new_vars_prefix = "A", dtc = QSDTC) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  mutate(
+    AVISIT = if_else(ADT <= TRTSDT, "BASELINE", VISIT),
+    AVISITN = if_else(ADT <= TRTSDT, 0, VISITNUM)
+  ) 
+```
+
+If the numeric values of the answers in `QSSTRESN` values are not ready to be used for deriving
+scores and require transformation, it is recommended that `QSSTRESN` is kept in
+the ADaM dataset for traceability, and the transformed value is stored in
+`AVAL`, since that's what will be used for the score calculation.
+For the CoEQ item 6 has to be handled in such way:
```

Comment:
Should we specify why item 6 needs to be handled like this?

Reply:
@AndersAskeland  I added some text about why item 6 needs to be reversed. Please feel free to change the wording.

Reply:
Thanks! I think it make sense to have it. I've made a slightly rephrased version in my latest review :)